source/extensions/transport_sockets/alts/tsi_socket.cc

```diff
@@ -145,7 +145,7 @@ Network::PostIoAction TsiSocket::doHandshakeNextDone(NextResultPtr&& next_result
   if (read_error_ || (!handshake_complete_ && end_stream_read_)) {
     ENVOY_CONN_LOG(debug, "TSI: Handshake failed: end of stream without enough data",
                    callbacks_->connection());
-    return Network::PostIoAction::Close;
+    return Network::PostIoAction::CloseError;
```

Comment:
There are 2 remaining uses of Network::PostIoAction::Close; in this file when finding a read 0 during handshakes.  Would it make sense to also consider those cases CloseError?

Reply:
I changed one to `CloseError` and one to `CloseGraceful` (when the certificate validation failed, since that's not and an I/O error).

Reply:
@ggreenway 

I guess the question is wherever or it is worth having the extra complexity that comes from having 2 close types when out of the 3 remaining uses 2 of them are handshake errors.  I guess it's fine to use CloseGraceful for the 3 remaining cases.